components/renku_data_services/notebooks/api/amalthea_patches/init_containers.py

```diff
@@ -79,21 +80,170 @@ def git_clone(server: "UserServer") -> list[dict[str, Any]]:
             "value": str(Path(etc_cert_volume_mount[0]["mountPath"]) / "ca-certificates.crt"),
         },
     ]
-    if (
-        isinstance(server.user, RegisteredUser)
-        and isinstance(server.user.gitlab_user, CurrentUser)
-        and not user_is_anonymous
-    ):
-        env += [
-            {"name": f"{prefix}USER__EMAIL", "value": server.user.gitlab_user.email},
+    if server.user.is_authenticated:
+        if server.user.email:
+            env.append(
+                {"name": f"{prefix}USER__EMAIL", "value": server.user.email},
+            )
+        if server.user.full_name:
+            env.append(
+                {
+                    "name": f"{prefix}USER__FULL_NAME",
+                    "value": server.user.full_name,
+                },
+            )
+        elif server.user.first_name or server.user.last_name:
+            full_name = server.user.first_name or ""
+            if full_name == "":
+                full_name = server.user.last_name or ""
+            else:
+                full_name += " " + (server.user.last_name or "")
+            env.append(
+                {
+                    "name": f"{prefix}USER__FULL_NAME",
+                    "value": server.user.full_name,
+                },
+            )
```

Comment:
no more like `server.user.get_full_name()` and use that instead of duplicating the logic in the 2-3 places this is used.
But maybe an get_environ method would be nice as well in addition to that, if the environ is the same for the different containers.

Reply:
I added a `get_full_name` method on the APIUser that uses the code you suggested.